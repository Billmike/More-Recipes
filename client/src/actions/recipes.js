import axios from 'axios';
import {
  ADD_RECIPE,
  GET_RECIPES,
  GET_ONE_RECIPE,
  GET_USER_RECIPES,
  ADD_FAVORITE_RECIPE,
  FETCH_FAVORITE_RECIPES,
  EDIT_RECIPE,
  REMOVE_RECIPE,
  ADD_REVIEW,
  UPVOTE_RECIPE,
  DOWNVOTE_RECIPE
} from './types';
import './toastrConfig';

export const addRecipe = recipe => ({
  type: ADD_RECIPE,
  recipe
});

export const getAllRecipes = (recipes) => {
  return {
    type: GET_RECIPES,
    recipes,
  };
};

export const getOneRecipe = (recipe) => {
  return {
    type: GET_ONE_RECIPE,
    id: recipe.id,
    recipe
  };
};

export const getUserRecipe = (userRecipe) => {
  return {
    type: GET_USER_RECIPES,
    userRecipe,
  };
};

export const addFavorites = (favoriteRecipes) => {
  return {
    type: ADD_FAVORITE_RECIPE,
    favoriteRecipes
  };
};

export const fetchFavorites = (favoriteRecipes) => {
  return {
    type: FETCH_FAVORITE_RECIPES,
    favoriteRecipes,
  };
};

export const editRecipe = (id, updates) => ({
  type: EDIT_RECIPE,
  id,
  updates,
});

export const removeRecipe = ({ id } = {}) => ({
  type: REMOVE_RECIPE,
  id
});

export const addReview = (review) => {
  return {
    type: ADD_REVIEW,
    review,
  };
};

export const upVoteRecipe = (id) => {
  return {
    type: UPVOTE_RECIPE,
    id,
  };
};

export const downVoteRecipe = (id) => {
  return {
    type: DOWNVOTE_RECIPE,
    id,
  };
};

export const startGetAllRecipes = () => {
  return (dispatch) => {
    axios.get('http://localhost:8000/api/v1/recipes')
      .then((res) => {
        dispatch(getAllRecipes(res.data.recipeData));
      })
      .catch((error) => {
        console.log(error);
      });
  };
}

export const startAddRecipe = (recipeData = {}) => {
  return (dispatch) => {
    const {
      name = '',
      description = '',
      img_url = 'no-img-here',
      category = '',
      ingredients = '',
      instructions = ''
    } = recipeData;

    const recipe = {
      name,
      description,
      img_url,
      category,
      ingredients,
      instructions,
    };
    axios.post('http://localhost:8000/api/v1/recipes', recipe)
      .then(() => {
        toastr.success('Recipe added successfully.');
        dispatch(addRecipe({
          id: recipe.id,
          ...recipe
        }));
      });
  };
};

export const startGetUserRecipes = () => {
  return (dispatch) => {
    axios.get('http://localhost:8000/api/v1/users/recipes')
      .then((res) => {
        dispatch(getUserRecipe(res.data.userRecipe));
      })
      .catch((error) => {
        console.log(error);
      });
  };
};

export const startEditRecipe = (id, updates) => {
  return (dispatch) => {
    axios.put(`http://localhost:8000/api/v1/recipes/${id}/modify`, updates)
      .then(() => {
        toastr.success('Recipe edited successfully.');
        dispatch(editRecipe(id, updates));
      })
      .catch((error) => {
        console.log('_<_<_<_<_<__<_', error);
      });
  };
};

export const startRemoveRecipe = (id) => {
  return (dispatch) => {
    axios.delete(`http://localhost:8000/api/v1/recipes/${id}`)
      .then(() => {
        toastr.success('Recipe deleted successfully.');
        dispatch(removeRecipe(id));
      })
      .catch((error) => {
        console.log(error);
      });
  };
};

export const startGetOneRecipe = (id) => {
  return (dispatch) => {
    axios.get(`http://localhost:8000/api/v1/recipes/${id}`)
      .then((res) => {
        dispatch(getOneRecipe(res.data.recipeData));
      })
      .catch((error) => {
        console.log(error);
      });
  };
};

export const startAddFavoriteRecipes = (id) => {
  return (dispatch) => {
    axios.post(`http://localhost:8000/api/v1/recipes/${id}/favorites`)
      .then((res) => {
        toastr.success(res.data.message);
        return dispatch(addFavorites(res.data));
      })
      .catch((err) => {
        if (err.response.data
          .message === 'You need to be logged in to perform this action.') {
          toastr.error(err.response.data.message);
        } else if (err.response.data
          .message === 'Sorry, you cannot perform this action on your own recipe.') {
          toastr.error(err.response.data.message);
        }
      });
  };
};

export const startGetUserFavorites = (id) => {
  return (dispatch) => {
    axios.get(`http://localhost:8000/api/v1/users/${id}/favorites`)
      .then((res) => {
        dispatch(fetchFavorites(res.data));
      })
      .catch((err) => {
        console.log(err);
      });
  };
};

export const startUpvoteRecipe = (id) => {
  return (dispatch) => {
    axios.post(`http://localhost:8000/api/v1/recipes/${id}/votes/upvote`)
      .then((res) => {
        toastr.success(res.data.message);
      })
      .catch((err) => {
        if (err.response.data
          .message === 'You need to be logged in to perform this action.') {
          toastr.error(err.response.data.message);
        } else if (err.response.data
          .message === 'You already upvoted on this recipe') {
          toastr.warning(err.response.data.message);
        }
      });
  };
};

export const startDownVoteRecipe = (id) => {
  return (dispatch) => {
    axios.post(`http://localhost:8000/api/v1/recipes/${id}/votes/downvote`)
      .then((res) => {
        toastr.success(res.data.message);
      })
      .catch((err) => {
        if (err.response.data
          .message === 'You need to be logged in to perform this action.') {
          toastr.error(err.response.data.message);
        } else if (err.response.data
          .message === 'You already downvoted on this recipe') {
          toastr.warning(err.response.data.message);
        }
      });
  };
};

export const startAddReview = (id, data) => {
  return (dispatch) => {
    axios.post(`http://localhost:8000/api/v1/recipes/${id}/reviews`, data)
      .then((res) => {
        dispatch(addReview(res.data));
      })
      .catch((err) => {
        if (err.response.data
          .message === 'You need to be logged in to perform this action.') {
          toastr.error(err.response.data.message);
        } else if (err.response.data
          .message === 'You can\'t post an empty review. Please, enter a happy review for this recipe.') {
          toastr.warning(err.response.data.message);
        }
      });
  };
};
