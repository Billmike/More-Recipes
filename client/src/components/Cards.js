import React, { Component } from 'react';

class Cards extends Component {
	render () {
		return (
			<div>
			  <div className="card-group">
          <div className="card index-card">
            <a href="./template/recipe_detail.html"><img className="card-img-top" src="../src/assets/img/pizzza.jpg" alt="Card image cap" /></a>
            <div className="card-body">
              <a href="./template/recipe_detail.html"><h4 className="card-title">Pizza Pepperoni with mushroom toppings.</h4></a>
              <p className="card-text">This is a wider card with supporting text below as a natural lead-in to additional content. This card has even longer content than the first to show that equal height action.</p>
              <a href="./template/recipe_detail.html" className="btn border border-secondary rounded">45 <i className="fa fa-eye" aria-hidden="true"></i></a>
              <a href="#" className="btn border border-secondary rounded" data-toggle="modal" data-target="#exampleModal">  89 <i className="fa fa-heart" aria-hidden="true"></i></a> 
              <a href="./template/recipe_detail.html#comment" className="btn border border-secondary rounded"> 6 <i className="fa fa-comments" aria-hidden="true"></i></a>
            </div>
            <div className="card-footer">
              <small className="text-muted">Last updated 3 mins ago</small>
            </div>
          </div>
          <div className="card index-card">
            <a href="./template/recipe_detail.html"><img className="card-img-top" src="../src/assets/img/pizzza.jpg" alt="Card image cap" /></a>
            <div className="card-body">
              <a href="./template/recipe_detail.html"><h4 className="card-title">Pizza Pepperoni with mushroom toppings.</h4></a>
              <p className="card-text">This is a wider card with supporting text below as a natural lead-in to additional content. This card has even longer content than the first to show that equal height action.</p>
              <a href="./template/recipe_detail.html" className="btn border border-secondary rounded">180 <i className="fa fa-eye" aria-hidden="true"></i></a>
              <a href="#" className="border border-secondary rounded btn" data-toggle="modal" data-target="#exampleModal">  150 <i className="fa fa-heart" aria-hidden="true"></i></a> 
              <a href="./template/recipe_detail.html#comment" className="btn border border-secondary rounded"> 70 <i className="fa fa-comments" aria-hidden="true"></i></a>
            </div>
            <div className="card-footer">
              <small className="text-muted">Last updated 3 mins ago</small>
            </div>
          </div>
          <div className="card index-card">
            <a href="./template/recipe_detail.html"><img className="card-img-top" src="../src/assets/img/pizzza.jpg" alt="Card image cap" /></a>
            <div className="card-body">
              <a href="./template/recipe_detail.html"><h4 className="card-title">Pizza Pepperoni with mushroom toppings.</h4></a>
              <p className="card-text">This is a wider card with supporting text below as a natural lead-in to additional content. This card has even longer content than the first to show that equal height action. </p>
              <a href="./template/recipe_detail.html" className="border btn border-secondary rounded">67 <i className="fa fa-eye" aria-hidden="true"></i></a>
              <a href="#" className="border btn border-secondary rounded" data-toggle="modal" data-target="#exampleModal">  49 <i className="fa fa-heart" aria-hidden="true"></i></a> 
              <a href="./template/recipe_detail.html#comment" className="btn border border-secondary rounded"> 90 <i className="fa fa-comments" aria-hidden="true"></i></a>
            </div>
            <div className="card-footer">
              <small className="text-muted">Last updated 3 mins ago</small>
            </div>
          </div>
        </div>
		</div>
		)
	}
}

export default Cards;
