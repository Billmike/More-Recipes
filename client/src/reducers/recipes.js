const recipeDefaultState = [];

export default (state = recipeDefaultState, action) => {
  switch (action.type) {
    case 'ADD_RECIPE':
      return [
        ...state,
        action.recipe,
      ];
    default:
      return state;
  }
};
