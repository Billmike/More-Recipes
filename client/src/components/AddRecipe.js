import React from 'react';
import RecipesForm from './RecipesForm';
import { connect } from 'react-redux';
import { addRecipe } from '../actions/recipes';

const AddRecipe = (props) => {
    console.log(props)
	return (
	   <div>
       <h1 className="container">Add Recipe</h1>
        <RecipesForm onSubmit={(recipe) => {
            props.dispatch(addRecipe(recipe));
            props.history.push('/')
        }} />
       </div>	
	)
}

export default connect()(AddRecipe);
