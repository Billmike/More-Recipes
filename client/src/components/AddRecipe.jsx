import React, { Component } from 'react';
import RecipesForm from './RecipesForm';
import { connect } from 'react-redux';
import { addRecipe } from '../actions/recipes';

export class AddRecipe extends Component {
  onSubmit = (recipe) => {
    this.props.addRecipe(recipe);
    this.props.history.push('/dashboard');
  };

  render() {
    return (
      <div>
        <h1 className="container"> Add Recipe </h1>
        <RecipesForm
          onSubmit={this.onSubmit}
        />
      </div>
      );
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    addRecipe: (recipe) => dispatch(addRecipe(recipe))
  }
}

// const AddRecipe = (props) => {
//     console.log(props)
// 	return (
// 	   <div>
//        <h1 className="container">Add Recipe</h1>
//         <RecipesForm onSubmit={(recipe) => {
//             props.dispatch(addRecipe(recipe));
//             props.history.push('/')
//         }} />
//        </div>	
// 	)
// }

export default connect(undefined, mapDispatchToProps)(AddRecipe);
