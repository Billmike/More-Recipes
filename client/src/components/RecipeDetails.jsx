import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import Emoji from 'react-emoji-render';
import Footer from './Footer';
import ReviewForm from './ReviewForm';
import Review from './Review';
import { getUserinfo } from '../actions/signinRequest';
import {
  startGetOneRecipe,
  startUpvoteRecipe,
  startDownVoteRecipe,
  startAddReview,
  startAddFavoriteRecipes
} from '../actions/recipes';
import pizza from '../assets/img/pancakes.jpeg';
import hearty from '../assets/img/giphy.gif';

class RecipeDetail extends Component {
  constructor(props) {
    super(props);
    this.state = {
      reviewText: '',
      reviews: [],
      recipe: {},
      upVotes: 0
    };
    this.upVoteRecipe = this.upVoteRecipe.bind(this);
    this.downVoteRecipe = this.downVoteRecipe.bind(this);
    this.onReviewFormChange = this.onReviewFormChange.bind(this);
    this.reviewRecipe = this.reviewRecipe.bind(this);
    this.favoriteRecipes = this.favoriteRecipes.bind(this);
  }

  componentDidMount() {
    this.props.startGetOneRecipe(this.props.match.params.id);
    this.props.getUserinfo();
  }

  // componentDidUpdate() {
  //   this.props.startGetOneRecipe(this.props.match.params.id);
  // }
  // componentWillReceiveProps(nextProps) {
  //   console.log('will reci====>', nextProps.recipe);
  //   const recipe = (nextProps.recipe) ? (nextProps.recipe) : {};
  //   const upVotes = (recipe.upVote) ? recipe.upVote : 0;
  //   this.setState({ recipe, upVotes })
  // }

  onReviewFormChange(event) {
    event.preventDefault();
    this.setState({
      reviewText: event.target.value
    });
  }

  favoriteRecipes(event) {
    event.preventDefault();
    this.props.startAddFavoriteRecipes(this.props.match.params.id);
  }

  reviewRecipe(event) {
    event.preventDefault();
    this.props.startAddReview(this.props.match.params.id, {
      content: this.state.reviewText
    });
    document.getElementById('form').reset();
  }

  upVoteRecipe(event) {
    event.preventDefault();
    this.props.startUpvoteRecipe(this.props.match.params.id);
  }

  downVoteRecipe(event) {
    event.preventDefault();
    this.props.startDownVoteRecipe(this.props.match.params.id);
  }

  render() {
    let reviews;
    let splitIngredients;
    let splitInstructions;
    if (this.props.recipe) {
      splitIngredients = this.props.recipe.ingredients
        .split('\n')
        .map((ingredient, index) => {
          return (
              <p className="indie-key" key={index}>
                {ingredient}
              </p>
          );
        });

      splitInstructions = this.props.recipe.instructions
        .split('\n')
        .map((instruction, index) => {
          return (
              <p className="indie-key" key={index}>
                {instruction}
              </p>
          );
        });

      reviews =
        this.props.recipe.reviews.length > 0 ? (
          this.props.recipe.reviews.map(review => {
            return (
              <div className="review-div">
                <Review
                  key={review.id}
                  review={review}
                  user={this.props.user.username}
                />
              </div>
            );
          })
        ) : (
          <p className="center-emoji">
            {' '}
            No reviews for this recipe yet. Want to be the first to review this?<Emoji text="B-)" />
          </p>
        );
    }
    return (
      <div>
        <div>
        <h4 className="recipe-detail-name">
          {this.props.recipe.name}
          </h4>
          <div className="recipe-detail-category">
            Category: {this.props.recipe.category}
          </div>
          <div className="container image-name">
            <div className="img-fluid">
              <img src={this.props.recipe.imageUrl} className=" recipe-detail-image" />
            </div>
            <div className="recipe-detials-btn">
                <Link
                  to="/"
                  className="btn details-action-button thumbs-up brown-thumb"
                  onClick={this.upVoteRecipe}
                >
                  {this.props.recipe.upVote}{' '}
                  <i className="fa fa-thumbs-up fa" aria-hidden="true" />
                </Link>
                <Link
                  to="/"
                  className="btn details-action-button brown-thumb"
                  onClick={this.downVoteRecipe}
                >
                  {' '}
                  {this.props.recipe.downVote}{' '}
                  <i className="fa fa-thumbs-down fa" aria-hidden="true" />
                </Link>
                <Link
                  to="/"
                  onClick={this.favoriteRecipes}
                  className="btn details-action-button heart-button"
                >
                  <i className="fa fa-heart my-heart" aria-hidden="true" />
                </Link>
              </div>
            <div className="">
              <div>
                <h4 className="recipe-detail-description"> Description </h4>
                <p className="recipe-description-paragraph">{this.props.recipe.description}</p>
              </div>
            </div>
          </div>
          <div>
          <div className="row container">
            <div className="col-6 ingredients-div">
              <h4 className="detail-title"> Ingredients </h4>
              {splitIngredients}
            </div>
            <div className="col-6 ingredients-div">
              <h4 className="detail-title"> Instructions </h4>
              {splitInstructions}
            </div>
          </div>
          </div>
          <div>
            <div>
              <ReviewForm
                onChange={this.onReviewFormChange}
                reviewText={this.state.reviewText}
                reviewRecipe={this.reviewRecipe}
              />
            </div>
          </div>
          <br />
          <div>
            <h6 className="text-center form-h6">
              What people are saying about this recipe
            </h6>
          </div>
          <div className="review-back">{reviews}</div>
        </div>
        <Footer />
      </div>
    );
  }
}

const mapStateToProps = (state, props) => {
  console.log('single', state);
  return {
    recipe: state.recipes.singleRecipe,
    upVote: state.recipes.singleRecipe.upVote,
    user: state.auth.userDetails
  };
};

export default connect(mapStateToProps, {
  getUserinfo,
  startGetOneRecipe,
  startUpvoteRecipe,
  startDownVoteRecipe,
  startAddReview,
  startAddFavoriteRecipes
})(RecipeDetail);
