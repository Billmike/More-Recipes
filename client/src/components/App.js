import React, { Component } from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import Navbar from './Navbar';
import Homepage from './Hompage';
import Dashboard from './Dashboard';
import Favorite from './Favorite';
import AddRecipe from './AddRecipe';
import EditRecipe from './EditRecipe';
import ProfilePage from './ProfilePage';
import NotFoundPage from './NotFoundPage';

class App extends Component {
    render() {
        return(
            <div>
                <BrowserRouter>
                    <div>
                        <Navbar />
                        <Switch>
                        <Route exact path="/" component={Homepage} />
                        <Route path="/dashboard" component={Dashboard} />
                        <Route path="/favorite" component={Favorite} />
                        <Route path="/add" component={AddRecipe} />
                        <Route path="/edit" component={EditRecipe} />
                        <Route path="/profile" component={ProfilePage} />
                        <Route component={NotFoundPage} />
                    </Switch>
                    </div>
                </BrowserRouter>
            </div>
        )
    }
}

export default App;
