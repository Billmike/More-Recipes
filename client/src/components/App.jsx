import React from 'react';
import { BrowserRouter, browserHistory, Route, Switch } from 'react-router-dom';
import Navbar from './Navbar';
import Homepage from './Hompage';
import Dashboard from './Dashboard';
import RecipeDetails from './RecipeDetails';
import Favorite from './Favorites';
import AddRecipe from './AddRecipe';
import EditRecipe from './EditRecipe';
import ProfilePage from './ProfilePage';
import LoginPage from './LoginPage';
import RegistrationPage from './RegistrationPage';
import RequireAuth from '../utils/RequireAuth';
import NotFoundPage from './NotFoundPage';

const App = () => (
  <div>
    <BrowserRouter history={browserHistory}>
      <div>
        <Navbar />
        <Switch>
          <Route exact path="/" component={Homepage} />
          <Route path="/dashboard" component={RequireAuth(Dashboard)} />
          <Route path="/favorites" component={RequireAuth(Favorite)} />
          <Route path="/add" component={RequireAuth(AddRecipe)} />
          <Route path="/edit/:id" component={RequireAuth(EditRecipe)} />
          <Route path="/recipe/:id" component={RecipeDetails} />
          <Route path="/profile" component={RequireAuth(ProfilePage)} />
          <Route path="/login" component={LoginPage} />
          <Route path="/register" component={RegistrationPage} />
          <Route component={NotFoundPage} />
        </Switch>
      </div>
    </BrowserRouter>
  </div>
);

export default App;
