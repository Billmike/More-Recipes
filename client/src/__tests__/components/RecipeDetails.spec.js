import React from 'react';
import { shallow } from 'enzyme';
import { RecipeDetail } from '../../components/RecipeDetails';
import recipes from '../fixtures/recipes';

describe('<RecipeDetails Component', () => {
  it('Should render the recipe details component properly', () => {
    const GetOneRecipeAction = jest.fn();
    const match = {
      params: {
        id: 1
      }
    };
    const wrapper = shallow(<RecipeDetail
      GetOneRecipeAction={() => Promise.resolve()}
      match={match} recipe={recipes[2]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
