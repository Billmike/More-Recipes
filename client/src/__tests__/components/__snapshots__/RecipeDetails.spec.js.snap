// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeDetails Component Should render the recipe details component properly 1`] = `
<div>
  <div>
    <h4
      className="recipe-detail-name"
    >
      Amazing fried rice
    </h4>
    <div
      className="recipe-detail-category"
    >
      Category: 
      Lunch
    </div>
    <div
      className="container image-name"
    >
      <div
        className="img-fluid center-image"
      >
        <img
          className="rounded recipe-detail-image"
        />
      </div>
      <div
        className="recipe-detials-btn"
      >
        <Link
          className="btn details-action-button thumbs-up brown-thumb"
          onClick={[Function]}
          replace={false}
          to="/"
        >
           
          <i
            aria-hidden="true"
            className="fa fa-thumbs-up fa"
          />
        </Link>
        <Link
          className="btn details-action-button thumbs-down"
          onClick={[Function]}
          replace={false}
          to="/"
        >
           
           
          <i
            aria-hidden="true"
            className="fa fa-thumbs-down fa"
          />
        </Link>
        <Link
          className="btn details-action-button heart-button"
          onClick={[Function]}
          replace={false}
          to="/"
        >
          <i
            aria-hidden="true"
            className="fa fa-heart my-heart"
          />
        </Link>
      </div>
      <div
        className=""
      >
        <div>
          <h4
            className="recipe-detail-description"
          >
            Description
          </h4>
          <p
            className="recipe-description-paragraph"
          >
            A really amazing fried rice
          </p>
        </div>
      </div>
    </div>
    <div>
      <div
        className="container"
      >
        <div
          className="ingredients-div"
        >
          <h4
            className="detail-title"
          >
             Ingredients 
          </h4>
          <hr />
          <li
            className="split-items"
            key="1"
          >
            Bread
          </li>
          <li
            className="split-items"
            key="1"
          >
            Samolina
          </li>
          <li
            className="split-items"
            key="1"
          >
            Beancake
          </li>
        </div>
        <div
          className="ingredients-div"
        >
          <h4
            className="detail-title"
          >
             Instructions 
          </h4>
          <hr />
          <li
            className="split-items"
            key="1"
          >
            Cook it really well
          </li>
          <li
            className="split-items"
            key="1"
          >
            Avoid mixing with water
          </li>
        </div>
      </div>
    </div>
    <div>
      <p
        className="auth-review"
      >
        You have to be Logged in to post a review
      </p>
    </div>
    <br />
    <div>
      <h6
        className="text-center form-h6"
      >
        Reviews
      </h6>
    </div>
    <div
      className="review-back"
    >
      <div
        className="review-div"
      >
        <Review
          key="1"
          review={
            Object {
              "content": "Amazing recipe you got",
              "createdAt": "19th-Jan-2018",
              "id": 1,
              "updatedAt": "19th-Jan-2018",
            }
          }
        />
      </div>
    </div>
  </div>
  <Footer />
</div>
`;
